---
layout: page
title: Programming 
permalink: /research/Postdoc_data_analysis/
---

## Progamming

I see programming as describing a wide variety of pursuits, and as such I like to switch between languages quite a bit. In my research I have used short small scripts to speed up repetitive tasks, typically I focus on using perl, python and rarely ruby. I also sometimes develop or help develop more complex and structured programs, an example of this would be phyx, and for phyx I used C++ because even though the compiler has the least helpful errors of all time. C++ is the right language as many of the programs in phyx are things I use on a weekly basis. Finally, I tend to use languages such as R for statistics and making figures. Whenever, I'm mentoring someone who has shown interest in programming I make sure they know there is more than one language and more than one purpose. Programming isn't just an intimidating world of complex algorithms, sometimes it can be used for trivial tasks that make life easier.

### Current Projects

_The following two projects are the ones I'm mainly focusing on at the moment. They've been invited for a special issue of SysBio so be on the lookout for them their._

[**PHylogenetic Analysis Into Lineages**](https://github.com/jfwalker/PHAIL): This is a method of pulling the multiple trees out of the data and builds off of the edge-based analysis procedures.

[**Branch Estimation Synthesizer**](https://github.com/jfwalker/BES): This is a method of putting branch lengths on that are informed by the gene trees. In short it is designed to only analyze edges that are concordant with the species tree and infer the branch lengths for the species tree using those values. It fulfills the niche of adding molecular branch lengths to summary methods, and can do it in a fraction of the time other methods can. The program is named after this [guy](https://en.wikipedia.org/wiki/Bes).

### Past projects
Some examples of work where a heavy focus has been on the program are:

[**phyx: phylogenetic tools for unix**](https://github.com/FePhyFoFum/phyx): This is a set of C++ programs designed to help with a wide variety of tasks in phylogenetics.

[**PyPHLAWD**](https://github.com/FePhyFoFum/PyPHLAWD): This python program allows the user to construct large phylogenetic trees through a hierachical clustering procedure. This same procedure can be found in the [Cacti to Carnivores paper](https://bsapubs.onlinelibrary.wiley.com/doi/full/10.1002/ajb2.1069), and can be done on transcriptome data using the less cleverly named [CombineClades.py](https://github.com/jfwalker/Clustering) program. PyPHLAWD also has a [website](https://fephyfofum.github.io/PyPHLAWD/) and it is no coincidence that website looks very similar to this one.

[**MGWE: Maximum GeneWise Edge**](https://github.com/jfwalker/MGWE): This is a perl program to calculate the likelihood of an edge as opposed to the tree as a whole.

For a longer list of programs I have written please see my [github](https://github.com/jfwalker).
